src/components/user-settings.tsx

```diff
@@ -162,11 +162,27 @@ export default function UserSettings() {
 	} );
 
 	const onRemoveSnapshots = useCallback( async () => {
-		if ( snapshots?.length === 0 ) {
+		if ( ! allSnapshots || allSnapshots.length === 0 ) {
 			return;
 		}
-		await deleteAllSnapshots( snapshots );
-	}, [ snapshots, deleteAllSnapshots ] );
+
+		const CANCEL_BUTTON_INDEX = 0;
+		const DELETE_BUTTON_INDEX = 1;
+
+		const { response } = await getIpcApi().showMessageBox( {
+			type: 'warning',
+			message: __( 'Delete all demo sites' ),
+			detail: __(
+				'All demo sites databases that exist for your WordPress.com account, along with all posts, pages, comments, and media, will be lost.'
+			),
+			buttons: [ __( 'Cancel' ), __( 'Delete all' ) ],
+			cancelId: CANCEL_BUTTON_INDEX,
+		} );
+
+		if ( response === DELETE_BUTTON_INDEX ) {
+			await deleteAllSnapshots( allSnapshots );
```

Comment:
What do you think about confirming the action to the user in a similar way we use when the user updates the demo site?